org.envirocar.app/src/org/envirocar/app/views/carselection/CarSelectionListAdapter.java

```diff
@@ -206,8 +209,12 @@ public void onClick(DialogInterface dialog, int i) {
      */
     protected void addCarItem(Car car) {
         this.add(car);
-        if(this.getCount() == 1) {
-            this.mSelectedCar = car;
+        if (mSelectedCar!=null) {
+            if (this.getCount() == 1) {
+                mSelectedCar = car;
+            } else {
+                mSelectedCar = mSelectedTempCar;
```

Comment:
Actually, this will effect in that the check below always will be false:
https://github.com/enviroCar/enviroCar-app/blob/9ab8fac01f971a1963d63aaa8dc7485c642cf1f5/org.envirocar.app/src/org/envirocar/app/views/carselection/CarSelectionListAdapter.java#L127

The result is, that when you restart the app, no car is selected initially. I'd suggest to add another clause to the count check, which proofs if `mSelectedCar` is `null`. If you do so, there is no need for the `mSelectedTempCar` variable.

Reply:
> Actually, this will effect in that the check below always will be false:
> 
> https://github.com/enviroCar/enviroCar-app/blob/9ab8fac01f971a1963d63aaa8dc7485c642cf1f5/org.envirocar.app/src/org/envirocar/app/views/carselection/CarSelectionListAdapter.java#L127
> 
> The result is, that when you restart the app, no car is selected initially. I'd suggest to add another clause to the count check, which proofs if `mSelectedCar` is `null`.

Sir but App is running perfectly and I have added the code that you mention.
I Checked that code will use when multiple cars are present and the user selects or changes the car then It will be used to deselect the car that was already selected.

https://github.com/enviroCar/enviroCar-app/blob/9ab8fac01f971a1963d63aaa8dc7485c642cf1f5/org.envirocar.app/src/org/envirocar/app/views/carselection/CarSelectionListAdapter.java#L132

Reply:
> 



> Actually, this will effect in that the check below always will be false:
> 
> https://github.com/enviroCar/enviroCar-app/blob/9ab8fac01f971a1963d63aaa8dc7485c642cf1f5/org.envirocar.app/src/org/envirocar/app/views/carselection/CarSelectionListAdapter.java#L127
> 
> The result is, that when you restart the app, no car is selected initially. I'd suggest to add another clause to the count check, which proofs if `mSelectedCar` is `null`.

Sir Below Code Only true when `mSelectedCar` is `car`. `car` is Dynamic 

https://github.com/enviroCar/enviroCar-app/blob/9ab8fac01f971a1963d63aaa8dc7485c642cf1f5/org.envirocar.app/src/org/envirocar/app/views/carselection/CarSelectionListAdapter.java#L127

Reply:
In addition, of course `mSelectedCar.equals(car)` would be `true` although `mSelectedCar` and `car` are different instances but have same properties, if `equal` would be overridden properly in the `Car` class. But that's not the point here.

Reply:
Sir, How can I solve it?

Reply:
> I'd suggest to add another clause to the count check, which proofs if mSelectedCar is null.

Give it a try. I've just posted it in one of the last comments.

Besides this, if you create a pull request that aims to fix an issue, you should come up with a proper solution for it. Endless review/revise cycles are not very effective and takes too much time, bot for you and me.

Reply:
> takes

Okay  And Sorry Sir @SebaDro